File: samples/C# Samples/DotNETTWAINDemo/DotNETTWAINDemo/RotateForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace DotNet_TWAIN_Demo
{
    public partial class RotateForm : Form
    {
        public RotateForm()
        {
            InitializeComponent();

            this.tbxAngle.Text = "45";
            this.cbxInterPolation.SelectedIndex = 1;
            this.cbxKeepSize.Checked = false;
            this.tbxR.Text = "255";
            this.tbxG.Text = "255";
            this.tbxB.Text = "255";

            this.DialogResult = DialogResult.Cancel;
        }

        private void btnCancel_Click(object sender, EventArgs e)
        {
            this.Close();
        }

        private void btnOK_Click(object sender, EventArgs e)
        {
            double angle;
            int r, g, b;

            if (!double.TryParse(tbxAngle.Text, out angle))
            {
                tbxAngle.Focus();
                return;
            }

            if (!int.TryParse(tbxR.Text, out r) || r < 0 || r > 255 )
            {
                tbxR.Focus();
                return;
            }

            if (!int.TryParse(tbxG.Text, out g) || g < 0 || g > 255)
            {
                tbxG.Focus();
                return;
            }

            if (!int.TryParse(tbxB.Text, out b) || b < 0 || b > 255)
            {
                tbxB.Focus();
                return;
            }
            
            this.DialogResult = DialogResult.OK;
            this.Close();
        }
    }
}

File: samples/VB .NET Samples/DotNETTWAINDemo/DotNETTWAINDemo/RotateForm.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Public Partial Class RotateForm
	Inherits Form
	Public Sub New()
		InitializeComponent()

		Me.tbxAngle.Text = "45"
		Me.cbxInterPolation.SelectedIndex = 1
		Me.cbxKeepSize.Checked = False
		Me.tbxR.Text = "255"
		Me.tbxG.Text = "255"
		Me.tbxB.Text = "255"

		Me.DialogResult = DialogResult.Cancel
	End Sub

	Private Sub btnCancel_Click(sender As Object, e As EventArgs)
		Me.Close()
	End Sub

	Private Sub btnOK_Click(sender As Object, e As EventArgs)
		Dim angle As Double
		Dim r As Integer, g As Integer, b As Integer

		If Not Double.TryParse(tbxAngle.Text, angle) Then
			tbxAngle.Focus()
			Return
		End If

		If Not Integer.TryParse(tbxR.Text, r) OrElse r < 0 OrElse r > 255 Then
			tbxR.Focus()
			Return
		End If

		If Not Integer.TryParse(tbxG.Text, g) OrElse g < 0 OrElse g > 255 Then
			tbxG.Focus()
			Return
		End If

		If Not Integer.TryParse(tbxB.Text, b) OrElse b < 0 OrElse b > 255 Then
			tbxB.Focus()
			Return
		End If

		Me.DialogResult = DialogResult.OK
		Me.Close()
	End Sub
End Class
